move_base/include/move_base/move_base.h

```diff
@@ -184,13 +184,14 @@ namespace move_base {
       std::string robot_base_frame_, global_frame_;
 
       std::vector<boost::shared_ptr<nav_core::RecoveryBehavior> > recovery_behaviors_;
+      std::vector<std::string> recovery_behavior_names_;
```

Comment:
It is frustrating (and necessary) that you have to do this because the `RecoveryBehavior` object doesn't have a toString method.